### PR TITLE
fix trash have not recycle wal bug

### PR DESCRIPTION
**Pull request by xu-chaojie**
### What problem does this PR solve?

Issue Number: close #xxx 

Problem Summary:

### What is changed and how it works?

What's Changed:

How it Works:

Side effects(Breaking backward compatibility? Performance regression?): 

### Check List

- [ ] Relevant documentation/comments is changed or added
- [x] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

File: src/chunkserver/trash.cpp [hunk 1]
```diff
@@ -264,6 +266,39 @@ bool Trash::RecycleIfChunkfile(
     return true;
 }
 
+bool Trash::RecycleWAL(
+    const std::string &filepath, const std::string &filename) {
+    if (0 != chunkFilePool_->RecycleFile(filepath)) {
```
Comment (wu-hanqing) [inline src/chunkserver/trash.cpp:271]: walPool？

File: test/chunkserver/trash_test.cpp [hunk 2]
```diff
@@ -372,6 +375,114 @@ TEST_F(TrashTest,
...
 }
 
+TEST_F(TrashTest, recycle_wal_success) {
+    std::vector<std::string> files{"4294967493.55555"};
+    std::vector<std::string> raftfiles{RAFT_LOG_DIR,
+        RAFT_SNAP_DIR, RAFT_META_DIR, RAFT_DATA_DIR, "hello"};
+    std::vector<std::string> logfiles{"curve_log_10086_10087",
+        "curve_log_inprogress_10088", "log_10083_10084", "log_inprogress_10085"};
+    std::vector<std::string> empty;
+        EXPECT_CALL(*lfs, DirExists(_))
```
Comment (wu-hanqing) [inline test/chunkserver/trash_test.cpp:385]: 缩进

File: test/chunkserver/trash_test.cpp [hunk 3]
```diff
@@ -372,6 +375,114 @@ TEST_F(TrashTest,
...
+}
+
+TEST_F(TrashTest, recycle_wal_failed) {
+    std::vector<std::string> files{"4294967493.55555"};
+    std::vector<std::string> raftfiles{RAFT_LOG_DIR,
+        RAFT_SNAP_DIR, RAFT_META_DIR, RAFT_DATA_DIR, "hello"};
+    std::vector<std::string> logfiles{"curve_log_10086_10087",
+        "curve_log_inprogress_10088", "log_10083_10084", "log_inprogress_10085"};
+    std::vector<std::string> empty;
+        EXPECT_CALL(*lfs, DirExists(_))
```
Comment (wu-hanqing) [inline test/chunkserver/trash_test.cpp:438]: 缩进

File: src/chunkserver/trash.h [hunk 4]
```diff
@@ -46,6 +46,7 @@ struct TrashOptions{
 
     std::shared_ptr<LocalFileSystem> localFileSystem;
     std::shared_ptr<FilePool> chunkFilePool;
+    std::shared_ptr<FilePool> walPool;
```
Comment (wu-hanqing) [inline src/chunkserver/trash.h:49]: 这里和trash内部，只使用了walpool，不是walpool的owner。保存一个WalPool指针就可以了。

File: src/chunkserver/trash.cpp (hunk 1 above)
Comment (xu-chaojie) [reply src/chunkserver/trash.cpp:271]: done

File: test/chunkserver/trash_test.cpp (hunk 2 above)
Comment (xu-chaojie) [reply test/chunkserver/trash_test.cpp:385]: done

File: test/chunkserver/trash_test.cpp (hunk 3 above)
Comment (xu-chaojie) [reply test/chunkserver/trash_test.cpp:438]: done

File: src/chunkserver/trash.h (hunk 4 above)
Comment (xu-chaojie) [reply src/chunkserver/trash.h:49]: 要保证在walpool回收时指针不能空悬，还是需要智能指针的

**Review by wu-hanqing — APPROVED** [review]

File: conf/chunkserver.conf.example [hunk 5]
```diff
@@ -167,6 +167,8 @@ chunkfilepool.retry_times=5
 #
 # WAL file pool
 #
+# walpool是否共用chunkfilepool，如果为true，则以下配置无效
+walfilepool.use_chunk_file_pool=true
```
Comment (ilixiaocui) [inline conf/chunkserver.conf.example:171]: 还有几个chunkserver的配置文件也一并改掉吧

Comment (xu-chaojie) [reply conf/chunkserver.conf.example:171]: done

File: src/chunkserver/trash.cpp (hunk 1 above)
Comment (ilixiaocui) [inline src/chunkserver/trash.cpp]: 这里为啥单测没能发现错误

Comment (xu-chaojie) [reply src/chunkserver/trash.cpp]: 因为单测也一块错了

File: conf/chunkserver.conf.example [hunk 6]
```diff
@@ -167,6 +167,8 @@ chunkfilepool.retry_times=5
 #
 # WAL file pool
 #
+# walpool是否共用chunkfilepool，如果为true，则以下配置无效
```
Comment (wu-hanqing) [inline conf/chunkserver.conf.example:170]: 这里相当于walpool有三种方式：
1. raft默认
2. walpool
3. chunkfilepool
如果第二种不好评估pool的容量，容易导致walpool中的chunk用完，是不是只保留1、3这两种就可以了？

Comment (xu-chaojie) [reply conf/chunkserver.conf.example:170]: 暂时还是保留walpool这种吧

Comment (ilixiaocui) [reply conf/chunkserver.conf.example:170]: 可以考虑去掉

**Review by wu-hanqing — COMMENTED** [review]
👍

**Review by ilixiaocui — APPROVED** [review]

**Pull request merged**